src/token_contract/src/main.nr

```diff
@@ -115,15 +119,15 @@ pub contract Token {
     ) {
         _validate_from_private(from, nonce, &mut context);
 
-        let token = Token::at(context.this_address());
+        let self = Token::at(context.this_address());
 
         // We prepare the private balance increase (the partial note).
         let hiding_point_slot =
             _prepare_transfer_public_to_private(from, to, &mut context, storage);
 
         // At last we finalize the transfer. Usage of the `unsafe` method here is safe because we set the `from`
         // function argument to a message sender, guaranteeing that he can transfer only his own tokens.
-        token._finalize_transfer_public_to_private_unsafe(from, amount, hiding_point_slot).enqueue(
+        self._finalize_transfer_public_to_private_unsafe(from, amount, hiding_point_slot).enqueue(
```

Comment:
we're redeclaring `self` inside `_prepare_transfer_public_to_private`, should we pass it as an argument?